src/Model/Cart.java

```diff
@@ -0,0 +1,81 @@
+package Model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import API.CartService;
+import Model.MailingList.EmailSender;
+import Model.paymentMethods.FinancialPayment;
+
+
+/**
+ * @author Jordan
+ * shopping cart
+ */
+public class Cart implements CartService{
+
+	protected List<Item> products;
+	private CustomerService userAccount;
+	
+	public Cart(CustomerService userAccount){
+		products=new ArrayList<Item>();
+		this.userAccount=userAccount;
+	}
+	
+	public void add(Product prod, int quantity){
+		products.add(new Item(quantity, prod));
+	}
+	
+	public Item removeLastItem(){
+		Item last=products.remove(products.lastIndexOf(products));
+		return last;
+	}
+	
+	public List<Item> getItems(){
+		return products;
+	}
+	
+	public void cleanCart(){
+		products=new ArrayList<Item>();
+	}
+	
+	/**
+	 * @param category
+	 * @return all items that contains a product with that category
+	 */
+	public List<Item> getProductsByCategory(String category){
+		ArrayList<Item> listByCategory=new ArrayList<Item>();
```

Comment:
This is just a consult, this gives you all the products (we are in a cart class) that have a category like the parameter. Why i need to retrieve data from my persistent data, if i already have in memory the actual state. Then when we are done we can persist the actual state.